client/my-sites/themes/theme-preview.jsx

```diff
@@ -2,79 +2,139 @@
+import { getThemeForPreviewData, getTheme } from 'state/themes/selectors';
+import { getPreviewUrl } from 'my-sites/themes/helpers';
+import { localize } from 'i18n-calypso';
+import { closePreview } from 'state/ui/preview/actions';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { isJetpackSite } from 'state/sites/selectors';
 
-export default React.createClass( {
-	displayName: 'ThemePreview',
+export default function themePreview( WebPreview ) {
+	const ThemePreview = React.createClass( {
+		displayName: 'ThemePreview',
 
-	propTypes: {
-		theme: React.PropTypes.object,
-		showPreview: React.PropTypes.bool,
-		showExternal: React.PropTypes.bool,
-		onClose: React.PropTypes.func,
-		primaryButtonLabel: React.PropTypes.string.isRequired,
-		onPrimaryButtonClick: React.PropTypes.func,
-		getPrimaryButtonHref: React.PropTypes.func,
-		secondaryButtonLabel: React.PropTypes.string,
-		onSecondaryButtonClick: React.PropTypes.func,
-		getSecondaryButtonHref: React.PropTypes.func,
-	},
+		propTypes: {
+			theme: React.PropTypes.object,
+			showPreview: React.PropTypes.bool,
+			showExternal: React.PropTypes.bool,
+			onClose: React.PropTypes.func,
+			getPrimaryButtonHref: React.PropTypes.func,
+			secondaryButtonLabel: React.PropTypes.string,
+			onSecondaryButtonClick: React.PropTypes.func,
+		},
 
-	getDefaultProps() {
-		return {
-			onPrimaryButtonClick: noop,
-			getPrimaryButtonHref: () => null,
-			onSecondaryButtonClick: noop,
-			getSecondaryButtonHref: () => null,
-		};
-	},
+		getDefaultProps() {
+			return {
+				getPrimaryButtonHref: () => null,
+				onSecondaryButtonClick: noop,
+			};
+		},
 
-	onPrimaryButtonClick() {
-		this.props.onPrimaryButtonClick( this.props.theme );
-		this.props.onClose();
-	},
+		onPrimaryButtonClick() {
+			const option = this.getPrimaryOption();
+			option.action && option.action( this.props.theme );
+			this.props.closePreview();
+		},
 
-	onSecondaryButtonClick() {
-		this.props.onSecondaryButtonClick( this.props.theme );
-		this.props.onClose();
-	},
+		onSecondaryButtonClick() {
+			const secondary = this.getSecondaryOption();
+			secondary.action && secondary.action( this.props.theme );
+			this.props.closePreview();
+		},
 
-	renderSecondaryButton() {
-		if ( ! this.props.secondaryButtonLabel ) {
-			return;
-		}
-		const buttonHref = this.props.getSecondaryButtonHref ? this.props.getSecondaryButtonHref( this.props.theme ) : null;
-		return (
-			<Button onClick={ this.onSecondaryButtonClick } href={ buttonHref } >
-				{ this.props.secondaryButtonLabel }
-			</Button>
-		);
-	},
+		getPrimaryOption() {
+			const { translate } = this.props;
+			const { purchase } = this.props.options;
+			const { price } = this.props.theme;
+			let { primary } = this.props.previewData.themeOptions;
+			primary.label = translate( 'Activate this design' );
+
+			if ( price && purchase ) {
+				primary = purchase;
+				primary.label = translate( 'Purchase this design' );
```

Comment:
Hi! I've found a possible matching string that has already been translated **31** times:
`translate( 'Purchase this theme' )` *ES Score: 6.36*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).